src/lib/y2storage/blk_device.rb

```diff
@@ -180,13 +194,35 @@ def filesystem_label
       blk_filesystem.label
     end
 
+    # full path of the udev by-label link or `nil` if it does not exist.
+    # e.g. "/dev/disk/by-label/DATA"
+    # @see #udev_paths
+    # @return [String]
+    def udev_full_label
+      label = filesystem_label
+
+      return nil unless label
```

Comment:
Well, you should check for empty label here. Take into account that attributes coming from libstorage-ng (like `filesystem#label`) are never nil. At most, they are empty.

Reply:
so it can be nil if there is no filesystem and can be also empty, which also means not defined, right?